tests/unit/data_processing/test_datetime_formatter.py

```diff
@@ -86,3 +86,57 @@ def test_format_datetime_does_not_match_format(self):
             '20220929111311000000',
         ])
         pd.testing.assert_series_equal(result, expected)
+
+    def test_format_data_datetime_dtype_returns_early(self):
+        """Test that if dtype is datetime64 and matches column, it returns early."""
+        # Setup
+        formatter = DatetimeFormatter('%Y-%m-%d')
+        column = pd.Series(pd.to_datetime(['2021-01-01', '2022-01-01']))
+        formatter._dtype = column.dtype
+
+        # Run
+        result = formatter.format_data(column)
+
+        # Assert
+        pd.testing.assert_series_equal(result, column)
+
+    def test_format_data_values_match_format_and_dtype(self):
+        """Test that if all values match the format and dtype matches, returns original."""
+        # Setup
+        column = pd.Series(['2023-01-01', '2023-01-02'])
+        formatter = DatetimeFormatter('%Y-%m-%d')
+        formatter._dtype = column.dtype  # Object
+
+        # Run
+        result = formatter.format_data(column)
+
+        # Assert
+        pd.testing.assert_series_equal(result, column)
+
+    def test_format_data_parsing_fallback_on_value_error(self):
+        """Test fallback formatting when parsing with format raises ValueError."""
+        # Setup
+        formatter = DatetimeFormatter('%Y/%m/%d')
+        formatter._dtype = 'O'
+        column = pd.Series(['2024-06-01', '2025-06-01'])  # does not match %Y/%m/%d
+
+        # Run
+        result = formatter.format_data(column)
+
+        # Assert
+        expected = pd.Series(['2024/06/01', '2025/06/01'])
+        pd.testing.assert_series_equal(result, expected)
+
+    def test_format_data_fallback_value_error_then_success(self):
+        """Test fallback parsing when initial format fails but fallback succeeds."""
+        # Setup
+        formatter = DatetimeFormatter('%Y/%m/%d')
+        formatter._dtype = 'O'
+        column = pd.Series(['2023-01-01', '2023-01-02'])  # doesn't match %Y/%m/%d
+
+        # Run
+        result = formatter.format_data(column)
+
+        # Assert
+        expected = pd.Series(['2023/01/01', '2023/01/02'])
+        pd.testing.assert_series_equal(result, expected)
```

Comment:
Out of curiosity, what is the difference between these two tests? I see they don't have the same column values, but I wasn't sure which case each one is checking.